pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -436,7 +436,8 @@ func (r *NSTemplateSetReconciler) ensureInnerNamespaceResources(logger logr.Logg
 // deleteRedundantObjects takes template objects of the current tier and of the new tier (provided as newObjects param),
 // compares their names and GVKs and deletes those ones that are in the current template but are not found in the new one.
 // return `true, nil` if an object was deleted, `false, nil`/`false, err` otherwise
-func (r *NSTemplateSetReconciler) deleteRedundantObjects(logger logr.Logger, currentTier, typeName, username string, newObjects []runtime.RawExtension) (bool, error) {
+func (r *NSTemplateSetReconciler) deleteRedundantObjects(logger logr.Logger, returnWhenDeleted bool, currentTier, typeName, username string, newObjects []runtime.RawExtension) (bool, error) {
```

Comment:
`returnWhenDeleted` is a bit confusing name. It's not about returning when deleted. It's rather about deleting the first object only or all of them, right? So, maybe something like `deleteFirstOnly` or opposite like `deleteAll` would be a better name?

Reply:
yeah, I know - I'm bad at naming things.
what about `deleteOnlyOne`?

Reply:
`deleteOnlyOne` looks good to me!

Reply:
renamed in 1d2bd7e